### ISSUE TITLE
Pydantic: Support for specifying field properties via Annotated[…, Field(…)]

### ISSUE THREAD

**Issue by gusutabopb** [astral-sh/ty#2130]
### Summary

Related to https://github.com/pydantic/pydantic/issues/6713

Currently ty doesn't detect that a field is optional in a Pydantic model constructor when using the `Annotated[<type>, <annotations>]` syntax, which is currently recommended way to add `Field` constraints.

Are there plans to support this? If not is there some other (not Pydantic-specific) annotation that could be used to tell ty the given parameter is not mandatory?

### Sample code

```python
from typing import Annotated

from pydantic import BaseModel, Field


class Model(BaseModel):
    name: str
    options: Annotated[list[str], Field(default_factory=list)]

m = Model(name="John")
```

### Error

```
error[missing-argument]: No argument provided for required parameter `options`
  --> main.py:10:5
   |
 8 |     options: Annotated[list[str], Field(default_factory=list)]
 9 |
10 | m = Model(name="John")
   |     ^^^^^^^^^^^^^^^^^^
   |
info: rule `missing-argument` is enabled by default

Found 1 diagnostic
```

### Workaround

The code below works fine.

```python
from pydantic import BaseModel, Field


class Model(BaseModel):
    name: str
    options: list[str] = Field(default_factory=list)


m = Model(name="John")
```

PS: I couldn't past a playground link because I wasn't able to add third-party packages (i.e. Pydantic)

### Version

ty 0.0.4 (c1e6188b1 2025-12-18)

**Comment by Viicos** [comment]
I'll note that there are some challenges in supporting this, e.g. if you specify multiple `Field()` functions within the `Annotated` metadata (and even combined with the assignment), iirc we don't clearly document which one takes priority right now. Order is also inconsistent and subject to change in V3 (see https://github.com/pydantic/pydantic/issues/10507).
